### ISSUE TITLE
Phantom assassin -> Phantom strike teleported WD to PA instead of PA to WD

### ISSUE THREAD

**Issue by nshikov** [ValveSoftware/Dota2-Gameplay#16057]
### Ability name

Phantom strike

### Description

Phantom assassin -> Phantom strike teleported WD to PA instead of PA to WD and that gave PA's team advantage

### Example Match ID (and possibly Timestamp)

7653537588 - 1:10:07

**Comment by yckmm** [comment]
Witch Doctor had mirror shield.

**Comment by nshikov** [comment]
Well, at first I thought it explains everything, so closed the issue, but then I I thought a bit more and it seems weird to behave this way, cause hero item should give them the advantage cause it's their item, but it's not.
Also for some reason Anti-mage `Counterspell` behaves differently and it blocks this spell, I would expect mirror shield to blocks this spells also.

**Comment by nshikov** [comment]
Well, at first I thought it explains everything, so closed the issue, but then I I thought a bit more and it seems weird to behave this way, cause hero item should give them the advantage cause it's their item, but it's causing the confusion of the player who owns it instead.
Also for some reason Anti-mage `Counterspell` behaves differently and it blocks this spell(the only difference I see in description between them is - `Counterspell` blocks and sends spells back and `Mirror Shiled` blocks and reflects spells, which sound quite similar), I would expect mirror shield to blocks this spells also.

**Comment by yckmm** [comment]
Yes, counter spell does not reflect blink strikes for whatever reason (in my opinion it should).
